File: ois/ois20/round4/ois11_round4_solutions.tar(1)/time.cpp
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 * NOTE: it is recommended to use this even if you don't
 * understand the following code.
 */

#include <stdio.h>
#include <assert.h>

// constraints

// input data
int H0, M0, H1, M1;

int main() {
//  uncomment the following lines if you want to read/write from files
//  freopen("input.txt", "r", stdin);
//  freopen("output.txt", "w", stdout);

    assert(4 == scanf("%d%d%d%d", &H0, &M0, &H1, &M1));

    // insert your code here
    int start = H0*60 + M0;
    int end = H1*60 + M1;
    int delta = (end > start ? end-start : 24*60 - start + end);

    printf("%d %d\n", delta/60, delta%60); // print the result
    return 0;
}

File: ois/ois20/round4/ois11_round4_solutions.tar(1)/time.pas
{ NOTE: it is recommended to use this even if you don't understand the following code. }

{ input data }
var
    H0, M0, H1, M1 : longint;
    count : longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(H0, M0);
    readln(H1, M1);

    { insert your code here }
    count := (H1*60 + M1) - (H0*60 + M0);
    if (count < 0) then
        count := 24*60 + count;

    writeln(count div 60, ' ', count mod 60); { print result }
end.
